client/blocks/editor-checkout-modal/index.tsx

```diff
@@ -34,6 +34,24 @@ function fetchStripeConfigurationWpcom( args: Record< string, unknown > ) {
 	return fetchStripeConfiguration( args, wpcom );
 }
 
+function removeHashFromUrl(): void {
+	try {
+		const newUrl = window.location.hash
+			? window.location.href.replace( window.location.hash, '' )
+			: window.location.href + '';
+
+		window.history.replaceState( null, '', newUrl );
+	} catch {}
+
+	try {
+		// Modifying history does not trigger a hashchange event
+		// so we manually create one and use try/catch because
+		// IE11 does not include HashChange
+		const event = new HashChangeEvent( 'hashchange' );
+		window.dispatchEvent( event );
```

Comment:
You may not need this manual event part. Checkout relies on the hash (and the HashChange event) to jump between steps. If this is happening after checkout, you probably don't need to trigger an event (unless you do need it for some reason).

Reply:
I guess the other scenario would be if you were to close the checkout modal early. But it sounds like it may not be needed in that one either, now that I know it's not relied upon more widely.

Thanks for the insights @sirbrillig !